Realização do exercício

sql1freitas opened the pull request.

sql1freitas committed:
Realização do exercício

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1 +1,33 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name, health=100):
```
According to the checklist, you should use annotations. Add type annotations to the arguments and return value of the `__init__` method.
Also, according to the checklist, you should place each argument on a new line, including `self`, with proper indentation and formatting.

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1 +1,33 @@
+
+class Herbivore(Animal):
+    def hide(self):
+        self.hidden = not self.hidden
```
According to the checklist, you can change the `bool` value in one line.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Hey there! 👋 Thanks for your submission. There are a couple of things we need to address before we can approve this PR. Specifically, in `app/main.py`, the `__init__` method in the base class `Animal` should include a type annotation for the `name` parameter. Also, please place each argument on a new line, including `self`, with proper indentation and formatting within the `__init__` method. I know you can get this done! 👍

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

sql1freitas committed:
Anotações de tipo para todos os métodos e parâmetros.

Argumentos em linhas separadas, com indentação correta.

Comentários e organização limpos, seguindo o estilo PEP8.

sql1freitas committed:
Anotações de tipo para todos os métodos e parâmetros.

Argumentos em linhas separadas, com indentação correta.

Comentários e organização limpos, seguindo o estilo PEP8.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Great job! 🎉 The code is well-structured, meets all the requirements, and the previous issues have been resolved. Keep up the good work! 👍

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
